### PR TITLE
Fix display of shadowed variables while debugging

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Ensure that shadowed variables have a unique name so they are properly
displayed in the variables pane.  Each nested scope will enclose the
variable name in a pair of parentheses.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Microsoft/vscode-go/issues/1974

**Comment by jhendrixMSFT** [comment]
image

File: src/debugAdapter/goDebug.ts [hunk 1]
```diff
@@ -45,6 +45,14 @@ enum GoReflectKind {
...
 }
 
+enum GoVariableFlags {
+	VariableEscaped = 1,
+	VariableShadowed = 2,
+	VariableConstant = 4,
+	VariableArgument = 8,
+	VariableReturnArgument = 16
+}
+
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply src/debugAdapter/goDebug.ts]: It starts [here](https://github.com/go-delve/delve/blob/4c9a72e486f1f0d0c90ecede8415a871dced8117/service/api/types.go#L203), the magic is in `1 << (iota)`; `iota` will start at zero and each line is incremented by 1.  See the definition of `iota` [here](https://github.com/golang/go/wiki/Iota).

[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply src/debugAdapter/goDebug.ts]: I've relocated `GoVariableFlags`.

**Comment by ramya-rao-a** [comment]
Do you know if something changed with `gometalinter`? The corresponding tests are failing...

File: src/debugAdapter/goDebug.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply src/debugAdapter/goDebug.ts]: Correct, for the entirety of the `const` block.  In a new `const` block `iota` is reset to zero.

**Comment by jhendrixMSFT** [comment]
The error is coming from a dependent tool `errcheck`.  Here's the offending commit https://github.com/alecthomas/gometalinter/commit/f64e0e4b6a4bbabe18b0c66400fd164e2521cb01#diff-c15d9d82fc36434e5329fe18954eaa32

File: src/debugAdapter/goDebug.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramya-rao-a) [reply src/debugAdapter/goDebug.ts]: thats so cool

**Pull request merged**